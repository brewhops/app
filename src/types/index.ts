import { Moment } from 'moment';

export type Employee = {
  username: string;
};

export type Tank = {
  id?: number | string;
  name: string;
  status: string;
  in_use: boolean;
};

export type Action = {
  id: number;
  name: string;
  description: string;
};

export type Recipe = {
  id: number;
  name: string;
  airplane_code: string;
  yeast: number;
  instructions: object;
};

export type Batch = {
  id?: string | number;
  name: string;
  generation: number;
  volume: number;
  bright: number;
  recipe_id: number;
  tank_id?: string | number;
  started_on?: string;
  completed_on?: string;
};

export type Task = {
  id?: number;
  added_on?: string;
  completed_on?: string;
  assigned?: boolean;
  batch_id: number;
  action_id: number;
  employee_id?: number;
};

export type Version = {
  id?: number;
  sg: number;
  ph: number;
  abv: number;
  temperature: number;
  pressure: number;
  measured_on?: string | Moment;
  completed?: false | Moment;
  batch_id?: number;
};

export type BrewhopsCookie = {
  id: string | number;
  username: string;
  admin: boolean;
};

export type ActionUpdate = {
  action: {
    id: string | number | undefined;
    completed: boolean;
    assigned: boolean;
    employee: {
      id: string | number;
    };
  };
};

export type BatchUpdateOrCreate = ActionUpdate & Batch & Version;
