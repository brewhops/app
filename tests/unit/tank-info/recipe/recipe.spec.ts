import RecipeComponent from "@/components/tank-info/recipe.vue";
import { Recipe } from "@/types/index";
import { shallowMount, Wrapper } from "@vue/test-utils";

describe("recipe.vue", () => {
  let propsData: any;

  beforeAll(() => {
    const recipe: Recipe = {
      name: "Spicy Hops Beer",
      airplane_code: "SHB",
      yeast: 2,
      instructions: [{
        ingredient: "Spicy Hops",
        ratio: ".5",
      }],
    };

    propsData = { recipe, volume: 10 };
  });

  it("renders recipe component", () => {
    const wrapper = shallowMount(RecipeComponent, { propsData });
    expect(wrapper).toMatchSnapshot();
  });

  it("displays correctly computed weight", () => {
    const wrapper = shallowMount(RecipeComponent, { propsData });
    expect(wrapper.find("tbody :nth-child(3)").text()).toEqual("5");
  });

});
