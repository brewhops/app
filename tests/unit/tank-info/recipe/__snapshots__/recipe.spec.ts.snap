// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipe.vue renders recipe component 1`] = `
<div id="recipe">
  <h2>Spicy Hops Beer</h2>
  <table class="table">
    <thead>
      <tr>
        <th>Dry Hop/Adjunct</th>
        <th>Rate</th>
        <th>Weight</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Spicy Hops</td>
        <td>.5</td>
        <td>5</td>
      </tr>
    </tbody>
  </table>
</div>
`;
